srcalc/src/main/java/gov/va/med/srcalc/domain/variable/AbstractVariable.java

```diff
@@ -58,18 +60,31 @@ void setId(final int id)
         this.fId = id;
     }
     
-    @Override
-    @Transient
+    @Basic
+    @Column(
+    		name = "VARIABLE_KEY",
+    		length = KEY_MAX,
+    		nullable = false,
+    		unique = true)
     public String getKey()
     {
-        return getDisplayName();
+        return fKey;
+    }
+    
+    /**
+     * Sets the internal key of the variable
+     * @throws IllegalArgumentException if the given key is over 40 characters
+     */
+    public void setKey(final String key)
+    {
+    	this.fKey = Preconditions.requireWithin(key, KEY_MAX);
     }
     
     @Basic
     @Column(
             length = DISPLAY_NAME_MAX,
             nullable = false,
-            unique = true)   // for now, we use display name as a key, so don't allow dupes
+            unique = true)
```

Comment:
As long as nothing is looking up the variable based on the display name, we can drop the unique constraint as well.